ocpmodels/datasets/lmdb_dataset.py

```diff
@@ -57,11 +57,21 @@ def __init__(self, config, transform=None):
 
             self._keys, self.envs = [], []
             for db_path in db_paths:
-                self.envs.append(self.connect_db(db_path))
-                length = pickle.loads(
-                    self.envs[-1].begin().get("length".encode("ascii"))
-                )
-                self._keys.append(list(range(length)))
+                cur_env = self.connect_db(db_path)
+                self.envs.append(cur_env)
+
+                # Load and encode all keys in the LMDB
+                cur_keys = [
+                    f"{j}".encode("ascii")
+                    for j in range(cur_env.stat()["entries"])
+                ]
+
+                # Discard any key called "length" which was included for
```

Comment:
Can you add a line in the docstring saying that keys can be any string except `length` (since that's not obvious without looking at the implementation)?

Reply:
Added in the doc strings.